src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift

```diff
@@ -164,7 +178,61 @@ enum PPAnalyticsCollector {
 				riskLevelChangedComparedToPreviousSubmission: riskLevelChangedComparedToPreviousSubmission,
 				dateChangedComparedToPreviousSubmission: dateChangedComparedToPreviousSubmission
 			)
-			Analytics.collect(.riskExposureMetadata(.create(newRiskExposureMetadata)))
+			store?.currentENFRiskExposureMetadata = newRiskExposureMetadata
+			return
+		}
+		let newRiskExposureMetadata = RiskExposureMetadata(
+			riskLevel: riskLevel,
+			riskLevelChangedComparedToPreviousSubmission: riskLevelChangedComparedToPreviousSubmission,
+			mostRecentDateAtRiskLevel: mostRecentDateWithCurrentRiskLevel,
+			dateChangedComparedToPreviousSubmission: dateChangedComparedToPreviousSubmission
+		)
+		store?.currentENFRiskExposureMetadata = newRiskExposureMetadata
+	}
+	
+	private static func updateCheckinRiskExposureMetadata(_ checkinRiskCalculationResult: CheckinRiskCalculationResult) {
+		let riskLevel = checkinRiskCalculationResult.riskLevel
+		let riskLevelChangedComparedToPreviousSubmission: Bool
+		let dateChangedComparedToPreviousSubmission: Bool
+
+		// if there is a risk level value stored for previous submission
+		if store?.previousCheckinRiskExposureMetadata?.riskLevel != nil {
+			if riskLevel !=
+				store?.previousCheckinRiskExposureMetadata?.riskLevel {
+				// if there is a change in risk level
+				riskLevelChangedComparedToPreviousSubmission = true
+			} else {
+				// if there is no change in risk level
+				riskLevelChangedComparedToPreviousSubmission = false
+			}
+		} else {
+			// for the first time, the field is set to false
+			riskLevelChangedComparedToPreviousSubmission = false
+		}
+
+		// if there is most recent date store for previous submission
+		if store?.previousCheckinRiskExposureMetadata?.mostRecentDateAtRiskLevel != nil {
+			if checkinRiskCalculationResult.mostRecentDateWithCurrentRiskLevel !=
+				store?.previousCheckinRiskExposureMetadata?.mostRecentDateAtRiskLevel {
+				// if there is a change in date
+				dateChangedComparedToPreviousSubmission = true
+			} else {
+				// if there is no change in date
+				dateChangedComparedToPreviousSubmission = false
+			}
```

Comment:
dateChanged = checkinRiskCalculationResult.mostRecentDateWithCurrentRiskLevel !=				store?.previousCheckinRiskExposureMetadata?.mostRecentDateAtRiskLevel

Reply:
nice idea, i extended it so we save some more lines :)